### PR TITLE
Makes allowed tests a non-diff list, updates diff to handle dtypes

### PR DESCRIPTION
**Pull request by mruberry**
- Creates allowed_tests.py so list of allowed tests is easier to update.
- Updates test/common_device_type diff to filter unsupported dtypes (half). 

Per @dlibenzi suggestion, this will make it easier to edit the list of tests we let XLA run. This diff update will also filter test variants that use half types. As we start annotating PyTorch tests with their dtypes this should reduce the number of test failures (half tests will be skipped) and increase coverage (other dtype variants will be still be run).

### CODE REVIEW

File: torch_patches/X10-device_test.diff [hunk 1]
```diff
@@ -1,117 +1,35 @@
...
+
+
++import torch_xla
++assert torch_xla  # Silences Flake (unused import)
++
++# Acquires XLA metadata
++import os
++from runpy import run_path
++file_dir = os.path.dirname(os.path.abspath(__file__))
++xla_meta_path = file_dir + "/../xla/test/allowed_tests.py"
```
Comment (dlibenzi) [inline torch_patches/X10-device_test.diff]: If we want to upstream this, they might not like to have `xla` path bolted in here.
I think we are fine to pickup the "allowed test" file from the environment, and then we can set that env variable in our `run_tests.sh` runner.

File: torch_patches/X10-device_test.diff [hunk 2]
```diff
@@ -1,117 +1,35 @@
...
++
 +class XLATestBase(DeviceTypeTestBase):
 +    device_type = 'xla'
-+
-+    allowed_tests = {
-+        'test_addcdiv',
-+        'test_addcmul',
-+        'test_diagonal',
-+        'test_isinf',
-+        'test_clamp',
```
Comment (ailzhang) [inline torch_patches/X10-device_test.diff]: I fixed `temp_clamp` ;) This need to be added to the new list :D

File: torch_patches/X10-device_test.diff (hunk 1 above)
Comment (ailzhang) [reply torch_patches/X10-device_test.diff]: Btw this file is designed to be only touched when adding new devices which is rare. So we won't get conflicts often I assume. I feel we can live with patch for now until we improve our test coverage up to parity with Pytorch?

**Comment by dlibenzi** [comment]
IMHO, given that we seem to be continuously adding/enabling new tests, this is a better approach.
I don't see why to wait to do the right thing given its simplicity.

**Comment by mruberry** [comment]
Fair point on discovering the file. I'll fix it now.

File: torch_patches/X10-device_test.diff (hunk 2 above)
Comment (mruberry) [reply torch_patches/X10-device_test.diff]: Thanks Ailing!

**Comment by dlibenzi** [comment]
> IMHO, given that we seem to be continuously adding/enabling new tests, this is a better approach.
> I don't see why to wait to do the right thing given its simplicity.

To expand on this.
This infrastructure is basically two things.
 1. Machinery
 2. Config
The first part is OK living as code patch, while the second IMHO should be a config file (in our case a config file with Python format 😄 ).

File: torch_patches/X10-device_test.diff (hunk 1 above)
Comment (mruberry) [reply torch_patches/X10-device_test.diff]: run_tests.sh now exports the XLA_TEST_DIR environment variable.

**Comment by mruberry** [comment]
In addition to using an environment variable for discovery I also made the names more appropriate.

**Pull request merged**